cache/cache.go

```diff
@@ -26,6 +51,7 @@ type Cache interface {
 
 const (
 	BackendMemcached = "memcached"
+	BackendRedis     = "redis"
 )
 
 type BackendConfig struct {
```

Comment:
I would add Redis support here. We want Redis support in Mimir too (experimental at the beginning).

Reply:
Discussed on Slack. Miguel prefers to give it a try in GEM first. I'm OK.